LSPS3/README.md

```diff
@@ -0,0 +1,317 @@
+
+`opening_fee` is how much the LSP will charge to open a JIT channel, in
+millisatoshis.
+
+`opening_fee_valid_until` is a datetime up to which the offered
+`opening_fee` is valid.
+
+`opening_fee_promise` is an arbitrary string that proves to the LSP
+that it previously promised a specific `opening_fee` with a specific
+`eopning_fee_valid_until`.
+
+LSPs:
+
+* SHOULD use a cryptographically-verifiable method of generating
+  `opening_fee_promise`, such as a MAC of some serialization of the
+  `opening_fee` and the `opening_fee_valid_until`, then encoded in
+  hexadecimal or other text encoding.
+* MUST ensure the promise is no longer than 512 bytes.
+* SHOULD avoid characters outside of the printable ASCII range.
+
+Clients:
+
+* MUST abort this flow if `opening_fee_promise` exceeds 512 bytes.
+* MUST NOT parse or interpret this string, and only provide it
+  verbatim later.
+
+> **Rationale** By using a promise that must be remembered by the
+> client, the LSP does not have to remember what it offered via this
+> API, reducing state storage, especially if the client then decides
+> that the `opening_fee` is too high and ultimately decides not to
+> continue this flow.
+> By using a cryptographically-verifiable method, the LSP can ensure
+> that it will only honour `opening_fee`s it actually promised.
+> A size limit is imposed so that LSPs cannot burden clients with
+> unreasonable storage requirements.
+
+`min_payment_size` and `max_payment_size` are the limits of the
+payment size, inclusive.
+These are in millisatoshis.
+The payment size is the amount that the client will receive on the
+JIT Channel.
+
+The client MUST ignore any additional fields it does not recognize.
+
+> **Rationale** This allows future versions of this spec to add new
+> fields to this response, without causing problems for clients that
+> only support the current version.
+
+### 2.  Request JIT Channel
+
+The client constructs a request body for a `POST /lsp/jitchannel`
+request, depending on their need.
+
+Example `POST /lsp/jitchannel` request:
+
+```JSON
+{
+    "version": 1,
+    "opening_fee": "546000",
+    "opening_fee_valid_until": "2023-02-23T08:47:30.511Z",
+    "opening_fee_promise": "abcdefghijklmnopqrstuvwxyz",
+    "payment_size": "42000",
+    "client_node_id": "03d4e028a0d4a90868ec202ab684fb0085779defea9ca7553e06146557631eec20"
+}
+```
+
+`version` is the version of this spec that will be used for this
+interaction.
+
+`opening_fee`, `opening_fee_valid_until`, and `opening_fee_promise`
+are the parameters acquired from the previous step.
+Clients MUST copy them verbatim from some previous
+`GET /lsp/jitchannel` response.
+LSPs SHOULD check that the `opening_fee_promise` does in fact
+prove that it previously promised the specified `opening_fee` and
+`opening_fee_valid_until`.
+LSPs SHOULD check that the `opening_fee_valid_until` is not a
+past datetime.
+
+`payment_size` is the amount that the client wants to receive.
```

Comment:
The client can ask for a bigger channel than the first payment, so I propose to add two fields: additional_capacity and minimum_additional_capacity. The LSP MAY add additional_capacity but MUST add at least minimum_additional_capacity or return an error if it doesn't agree.

Reply:
Assuming we use a proportional feerate for `opening_fee`, should we consider `(additional_capacity + payment_size) / 0.99` or `(min_additional_capacity + payment_size) / 0.99` for `channel_size` that gets `opening_fee`?

Maybe it is cleaner to just have the client always specify `channel_size` instead, and just have the LSP check `payment_size / 0.99` is less than or equal to `channel_size`, as well as check if `channel_size` is too large.

Reply:
Perhaps instead of using hard coded 0.99, the formula has to be `opening_fee_base + opening_fee_proportional * (channel_size - channel_reserve)` . And both `channel_size` and `channel_reserve` has to be return to the client, so the client can also calculate the fees to know when to setttle the payment

Reply:
The reserve is really only relevant for the `payment_size` vs `channel_size` sanity check (it would be an error for the client to specify a channel size that cannot accept the initial payment size); the cost on the LSP is proportional on `channel_size` anyway, and the reserve is for the security of the client, so charging based on `channel_size` without removing the `channel_reserve`.

Reply:
I agree